Scripts/models/linear.py

```diff
@@ -0,0 +1,105 @@
+                data_capital_region = zdata.get_data(
+                    i, self.bounds, generation, zdata.CAPITAL_REGION)
+                data_surrounding = zdata.get_data(
+                    i, self.bounds, generation, zdata.SURROUNDING_AREA)
+                prediction[:k] += b[i][0] * data_capital_region
+                prediction[k:] += b[i][1] * data_surrounding
+        return prediction
+
+    def _add_log_zone_terms(self, prediction, b, generation=False):
+        zdata = self.zone_data
+        for i in b:
+            prediction += b[i] * numpy.log(zdata.get_data(i, self.bounds,
+                generation))
+        return prediction
+
+
+class CarDensityModel(LinearModel):
+    def predict(self):
+        b = parameters.car_density
+        prediction = numpy.zeros(self.bounds.stop)
+        self._add_constant(prediction, b["constant"])
+        self._add_zone_terms(prediction, b["generation"], True)
+        self._add_log_zone_terms(prediction, b["log"], True)
+        prediction = prediction.clip(0.0, None)
+        prediction = pandas.Series(
+            prediction, self.zone_data.zone_numbers[self.bounds])
+        self.print_results(prediction)
+        return prediction
+
+    def print_results(self, prediction):
+        """ Print results, mainly for calibration purposes"""
+        population = self.zone_data["population"]
+        population_7_99 = ( population[:self.zone_data.first_peripheral_zone]
+                          * self.zone_data["share_age_7-99"])
+        car_density = prediction
+                
+        # Print car density by zone
+        self.resultdata.print_data(
+            prediction, "car_density.txt",
+            self.zone_data.zone_numbers[self.bounds], "car_density")
+        
+        # print car density by municipality
+        prediction_municipality = []
+        for municipality in parameters.municipality:
+            i = slice(parameters.municipality[municipality][0],
+                      parameters.municipality[municipality][1])
+            # comparison data has car user shares of population
+            # over 6 years old (from HEHA)
+            prediction_municipality.append( car_density.loc[i].sum() 
+                                    / population_7_99.loc[i].sum())
+        self.resultdata.print_data(
+            prediction_municipality, "car_density_per_municipality.txt",
+            parameters.municipality.keys(), "car_density")
+                          
+        # print car density by area (to get Helsinki CBD vs. Helsinki other)
+        prediction_area = []
+        for area in parameters.areas:
+            i = slice(parameters.areas[area][0],
+                      parameters.areas[area][1])
+            # comparison data has car user shares of population
+            # over 6 years old (from HEHA)
+            prediction_area.append( car_density.loc[i].sum()
+                            / population_7_99.loc[i].sum())
```

Comment:
Sum of results are not reasonable. Should the numerator be product of car density and population by zone (weighted average)?

Reply:
Does this work? https://github.com/HSLdevcom/helmet-model-system/pull/120/commits/ad85e88d710e3cc92312236adfd5dc326e8e13c1